pacman_mirrors/pacman_mirrors.py

```diff
@@ -447,175 +608,368 @@ def random_servers(self, countries):
+
+        except (URLError, timeout, HTTPException, json.JSONDecodeError):
+            return []
+        # parse response from geo server
+        try:
+            # get geo country from response
+            country_name = json_obj["country_name"]
+        except KeyError:
+            return []
+        # dictionary
+        country_fix = {
+            "Brazil": "Brasil",
+            "Costa Rica": "Costa_Rica",
+            "Czech Republic": "Czech",
+            "South Africa": "Africa",
+            "United Kingdom": "United_Kingdom",
+            "United States": "United_States",
+        }
+        # decision on country
+        if country_name in country_fix.keys():
+            country_name = country_fix[country_name]
+        # return result
+        return country_name
+
+    @staticmethod
+    def get_mirror_branch_age(time_now, time_branch_state):
+        """
+        Calculates elapsed time since last syncronisation
+
+        :param: time_now: current time
+        :param: time_branch_state: date from state file
+        :return: elapsed_time
+        """
+        # calculate when the last update occurred compared to now
+        # calculate seconds elapsed since last update
+        total_seconds = (time_now - time_branch_state).total_seconds()
+        # convert to minutes
+        total_minutes = total_seconds // 60
+        # extract full elapsed_hours
+        elapsed_hours = total_minutes // 60
+        # extract remaining elapsed_minutes
+        elapsed_minutes = total_minutes % 60
+        # format last sync in format hh:mm
+        elapsed_time = "{}:{}".format(
+            int(elapsed_hours), str(int(elapsed_minutes)).zfill(2))
+        return elapsed_time
+
+    @staticmethod
+    def get_mirror_branch_state(state_url, mirror_branch, request_timeout, verbose=False):
+        """
+        Get statefile
+
+        :param: mirror_url
+        :return: content
+        """
+        # initialize content
+        content = ""
+        # find selected_branch position in url
+        position = state_url.find(mirror_branch)
+        # setup request
+        req = Request(state_url[0:position] + "state")
+        try:
+            # probe servers state file
+            with urlopen(req, timeout=request_timeout) as state_file:
+                # get state file content for later use
+                content = state_file.read()
+        # handle url error
+        except URLError as err:
+            # print error message to console
+            if hasattr(err, "reason"):
+                # log error reason to console
+                if verbose:
+                    print("\n" + _("Error: Failed to reach "
+                                   "the server: {reason}"
+                                   .format(reason=err.reason)))
+            elif hasattr(err, "code"):
+                # log error number to console
+                if verbose:
+                    print("\n" + _("Error: The server couldn\"t "
```

Comment:
The same
`Error: The server couldn"t ` is not correct grammar
Correct:
`Error: The server couldn't `